src/_pytest/fixtures.py

```diff
@@ -805,7 +840,7 @@ def formatrepr(self) -> FixtureLookupErrorRepr:
         stack = [self.request._pyfuncitem.obj]
         stack.extend(map(lambda x: x.func, self.fixturestack))
         msg = self.msg
-        if msg is not None:
+        if msg is not None and len(stack) > 1:
```

Comment:
This change does not seem necessary?

Reply:
ah yes, this was from an earlier version where I raised a `FixtureLookupError` - although I think the change might be good on its own because if `formatrepr` is called with `msg is not None and len(stack) <= 1` then the function will crash due to `fspath` never getting set. I could also change it to an assert instead
```suggestion
        if msg is not None:
            assert len(stack) > 1
```

Reply:
I might be missing something, but the only thing that happens inside this `if` is `stack[:-1]`, which cannot fail even if `stack` is empty.

Reply:
yes the problem is further down. If `len(stack) == 1` when entering this if statement then this loop isn't entered
https://github.com/pytest-dev/pytest/blob/283db4ed9f803c6b1f19b8920ed8b01cfda20b59/src/_pytest/fixtures.py#L847-L848
so `fspath` isn't set, and this line fails
https://github.com/pytest-dev/pytest/blob/283db4ed9f803c6b1f19b8920ed8b01cfda20b59/src/_pytest/fixtures.py#L880

I slightly misspoke saying `<= 1`, if the stack is empty then it doesn't matter if the if-statement is entered or not, but given that it's primed that can't happen

https://github.com/pytest-dev/pytest/blob/283db4ed9f803c6b1f19b8920ed8b01cfda20b59/src/_pytest/fixtures.py#L840-L841

Perhaps a more informative check/assert would be directly on `self.fixturestack`

Reply:
unless we go back to raising an error with the new implementation I'll separate this out to a separate PR

Reply:
A separate PR would make it easier to review and merge each part 👍 

(and we won't make this an error immediately)